### DISCUSSION TITLE
setFocus returns undefined

### DISCUSSION THREAD

**Discussion by beverlylaurel** [react-hook-form/react-hook-form#5212] [Q&A] [answered]
I want to trigger the focus event using my inputs using the setFocus method. but the error i got was "**setFocus is not a function**".

When I check setFocus by logging it, it looks undefined. where am I making a mistake?
  
Note: register and handleSubmit work fine.

```
 const { register, handleSubmit, setFocus } = useForm();
 console.log(setFocus);
```
  
 output: undefined

**Comment by bluebill1049** [comment, marked as answer]
which version?

**Reply by beverlylaurel** [reply]
@bluebill1049  thank you for the quick reply
  
  "react-hook-form": "^6.15.4",

**Reply by bluebill1049** [reply]
This function is v7 only.

**Reply by beverlylaurel** [reply]
thank you ✔

**Reply by aggarwal-muskaan** [reply]
Even in v7, although it's not throwing any error, setFocus is not focusing on the input.
@bluebill1049

**Reply by aewinformatica** [reply]
SetFocus in React native TextInput send error
TypeError: s.focus is not a function. (In 's.focus()', 's.focus' is undefined)

**Reply by AlexCernik** [reply]
setFocus not found on RN, is undefined.

**Reply by Moshyfawn** [reply]
A reproduction snack would really help ;)

Can't reproduce: see [this Snack](https://snack.expo.dev/@moshyfawn/react-hook-form)

P.S. Don't forget to register your `TextInput` `ref` in order for RHF to be able to focus your input
